services/github/github-directory-file-count.service.js

```diff
@@ -0,0 +1,105 @@
+'use strict'
+
+const path = require('path')
+const Joi = require('joi')
+const { InvalidParameter } = require('..')
+const { ConditionalGithubAuthV3Service } = require('./github-auth-service')
+const { documentation, errorMessagesFor } = require('./github-helpers')
+
+const schema = Joi.alternatives(
+  Joi.object({}).required(),
+  Joi.array()
+    .items(
+      Joi.object({
+        path: Joi.string().required(),
+        type: Joi.string().required(),
+      })
+    )
+    .required()
+)
+
+const queryParamSchema = Joi.object({
+  type: Joi.any().valid('dir', 'file'),
+  extension: Joi.string(),
+})
+
+module.exports = class GithubDirectoryFileCount extends ConditionalGithubAuthV3Service {
+  static category = 'size'
+
+  static route = {
+    base: 'github/directory-file-count',
+    pattern: ':user/:repo/:path*',
+    queryParamSchema,
+  }
+
+  static examples = [
+    {
+      title: 'GitHub repo directory file count',
+      pattern: ':user/:repo/:path',
+      namedParams: { user: 'badges', repo: 'shields', path: 'services' },
+      staticPreview: this.render({ count: 10 }),
+      documentation,
+    },
+    {
+      title: 'GitHub repo directory file count (dir type)',
+      pattern: ':user/:repo/:path',
+      namedParams: { user: 'badges', repo: 'shields', path: 'services' },
+      queryParams: { type: 'dir' },
+      staticPreview: this.render({ count: 8 }),
+      documentation,
+    },
+    {
+      title: 'GitHub repo directory file count (file type)',
+      pattern: ':user/:repo/:path',
+      namedParams: { user: 'badges', repo: 'shields', path: 'services' },
+      queryParams: { type: 'file' },
+      staticPreview: this.render({ count: 2 }),
+      documentation,
+    },
+    {
+      title: 'GitHub repo directory file count (file extension)',
+      pattern: ':user/:repo/:path',
+      namedParams: { user: 'badges', repo: 'shields', path: 'services' },
+      queryParams: { extension: 'js' },
+      staticPreview: this.render({ count: 1 }),
+      documentation,
+    },
+  ]
+
+  static defaultBadgeData = { color: 'blue', label: 'files' }
+
+  static render({ count }) {
+    return {
+      message: count,
```

Comment:
Let's use the `metric` helper (can be imported from `../text-formatters`) here to ensure proper and consistent formatting of large numbers. I'd hope there's not a ton of directories that have more than 1,000 files but I'm sure there's some out there :smile:

Reply:
Still want to use `metric(count)` for the value here. I suppose that alternatively we could check if count is equal to 1000 and conditionally change the value to `1k` instead, but would much rather use our existing helper to ensure consistency across the project